File: data-structure/Stack/Python/Stack.py
class Stack(object):
	def __init__(self):
		self.items = []

	def getSize(self):
		return self.len(items)

	def isEmpty(self):
		return getSize() == 0

	def push(self, data):
		self.items.append(data)

	def pop(self):
		self.items.pop()

	def peek(self):
		return self.items[getSize() - 1]

	def display(self):
		print "top"
		for i in self.items[::-1]:
			print i

File: data-structure/Stack/Java/Stack.java
class Stack {
	private int[] arr;
	private int size;
	private int top;

	public Stack(int size) {
		arr = new int[size];
		this.size = size;
		this.top = -1;
	}

	public int getSize() {
		return arr.length;
	}

	public boolean isEmpty() {
		return top == -1;
	}

	public void getTop() {
		System.out.println("top " + arr[top]);
	}

	public boolean isFull() {
		return top == size -1;
	}

	public void push(int data) {
		if(!isFull()) {
			top++;
			arr[top] = data;
		}
	}

	public void pop() {
		if (!isEmpty()) {
			int removed = arr[top--];
			System.out.println("popped element is " + removed);
		}
	}
}